dtm0/dtx.c

```diff
@@ -336,68 +343,90 @@ static void dtx_exec_all_ast_cb(struct m0_sm_group *grp, struct m0_sm_ast *ast)
 static void dtx_persistent_ast_cb(struct m0_sm_group *grp,
 				  struct m0_sm_ast   *ast)
 {
+	struct m0_dtm0_log_rec       *rec;
+	struct m0_dtm0_dtx           *dtx;
 	struct m0_dtm0_pmsg_ast      *pma = ast->sa_datum;
-	struct m0_dtm0_dtx           *dtx = pma->p_dtx;
 	struct m0_fop                *fop = pma->p_fop;
+	struct m0_be_dtm0_log        *log = pma->p_log;
 	struct dtm0_req_fop          *req = m0_fop_data(fop);
 	const struct m0_dtm0_tx_desc *txd = &req->dtr_txr;
 
-	M0_ENTRY("dtx=%p, fop=%p", dtx, fop);
+	M0_ENTRY("fop=%p", fop);
 
-	if (M0_IS0(dtx)) {
-		M0_LOG(M0_DEBUG, "Dtx has already reached DONE state, "
-		       "ignoring P msg for " DTID0_F ".",
-		       DTID0_P(&txd->dtd_id));
-		goto out;
-	}
-
-	m0_dtm0_tx_desc_apply(&dtx->dd_txd, txd);
-	dtx_log_update(dtx);
-
-	if (dtx->dd_sm.sm_state == M0_DDS_EXECUTED_ALL &&
-	    m0_dtm0_tx_desc_state_eq(&dtx->dd_txd, M0_DTPS_PERSISTENT)) {
-		M0_ASSERT(dtx->dd_nr_executed == dtx->dd_txd.dtd_ps.dtp_nr);
-		M0_LOG(M0_DEBUG, "dtx " DTID0_F "is stable (PMA)",
-		       DTID0_P(&dtx->dd_txd.dtd_id));
-		m0_sm_state_set(&dtx->dd_sm, M0_DDS_STABLE);
+	m0_mutex_lock(&log->dl_lock);
+	rec = m0_be_dtm0_log_find(log, &txd->dtd_id);
```

Comment:
I checked the current implementation of `m0_be_dtm0_log_find()` and it looks terrible performance-wise (always traversing the list of log records to find the needed one). Are there any plans to improve it?

Reply:
Yes, it will be replaced with new DTM0 log implementation.

Reply:
Please, add a comment about it then. Thanks.